src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -2534,7 +2546,104 @@ namespace winrt::TerminalApp::implementation
     {
         for (uint32_t i = 0; i < _tabs.Size(); ++i)
         {
-            _GetStrongTabImpl(i)->UpdateTabViewIndex(i);
+            _tabs.GetAt(i).UpdateTabViewIndex(i);
+        }
+    }
+
+    void TerminalPage::_OpenSettingsUI()
+    {
+        // If we're holding the settings tab's switch command, don't create a new one, switch to the existing one.
+        if (!_switchToSettingsCommand)
+        {
+            // Add the new tab to the list of our tabs.
+            auto newTabImpl = winrt::make_self<SettingsTab>();
+            _tabs.Append(*newTabImpl);
+
+            // Give the tab its index in the _tabs vector so it can manage its own SwitchToTab command.
+            newTabImpl->UpdateTabViewIndex(_tabs.Size() - 1);
+
+            // Don't capture a strong ref to the tab. If the tab is removed as this
+            // is called, we don't really care anymore about handling the event.
+            auto weakTab = make_weak(newTabImpl);
+
+            auto tabViewItem = newTabImpl->TabViewItem();
+            _tabView.TabItems().Append(tabViewItem);
+
+            // GH#6570
+            // The TabView does not apply compact sizing to items added after Compact is enabled.
+            // By forcibly reapplying compact sizing every time we add a new tab, we'll make sure
+            // that it works.
+            // Workaround from https://github.com/microsoft/microsoft-ui-xaml/issues/2711
+            if (_tabView.TabWidthMode() == MUX::Controls::TabViewWidthMode::Compact)
+            {
+                _tabView.UpdateLayout();
+                _tabView.TabWidthMode(MUX::Controls::TabViewWidthMode::Compact);
+            }
```

Comment:
_urgh_. XAML2711 is closed; I guess we haven't ingested the change yet? Is it possible for us to consolidate this into a helper function? That way we just remove it altogether when we don't need it anymore.

Reply:
Fine with me, I think it's only used here and in the function that creates a TerminalTab anyway